File: Apply Operations to an Array/Apply_Operations_to_an_Array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        n = len(nums)
        for i in range(n - 1):
            if nums[i] == nums[i + 1]:
                nums[i] *= 2
                nums[i + 1] = 0

        first, second = 0, 0
        while second < n:
            if nums[first] == 0 and nums[second] != 0:
                nums[first], nums[second] = nums[second], nums[first]
            if nums[first] != 0:
                first += 1
            second += 1
            
        return nums
        

File: Apply Operations to an Array/Apply_Operations_to_an_Array.cpp
class Solution {
public:
    vector<int> applyOperations(vector<int>& nums) {
        int n = nums.size();
        for(int i = 0; i<n-1; i++) {
            if(nums[i] == nums[i+1]) {
                nums[i] *= 2;
                nums[i+1] = 0;
            }
        }
        int first = 0,second = 0;
        while(second < n) {
            if(nums[first] == 0 && nums[second] != 0) {
                int temp = nums[first];
                nums[first] = nums[second];
                nums[second] = temp;
            }
            if(nums[first] != 0) first++;
            second++;
        }
        return nums;
    }
};